SEGFAULT in some erlang projects' build

Observed with `erlang-uuid`:

```
 🔸 Building «erlang-uuid» with firebuild, empty cache
Firebuild Git v0.1-401-g0533b13
...
make[1]: Entering directory '/home/ubuntu/perftest-build/erlang-uuid/erlang-uuid-0.4.6'
cp src/uuid.app.src ebin/uuid.app
erl -make
make[1]: *** [Makefile:18: build] Segmentation fault (core dumped)
make[1]: Leaving directory '/home/ubuntu/perftest-build/erlang-uuid/erlang-uuid-0.4.6'
...
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `/usr/lib/erlang/erts-12.2.1/bin/beam.smp -B -- -root /usr/lib/erlang -progname'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  __strstr_sse2_unaligned () at ../sysdeps/x86_64/multiarch/strstr-sse2-unaligned.S:40
40      ../sysdeps/x86_64/multiarch/strstr-sse2-unaligned.S: No such file or directory.
[Current thread is 1 (Thread 0x7f7980e6ebc0 (LWP 1130))]
#0  __strstr_sse2_unaligned () at ../sysdeps/x86_64/multiarch/strstr-sse2-unaligned.S:40
No locals.
#1  0x00005572def3516b in ?? ()
No symbol table info available.
#2  0x0000000000000001 in ?? ()
No symbol table info available.
#3  0x0000000000000001 in ?? ()
No symbol table info available.
#4  0x00005572dfe842c0 in ?? ()
No symbol table info available.
#5  0x0000000000000000 in ?? ()
No symbol table info available.
```
Seems like a stack corruption issue, but could not debug further.

--- a/src/common/fbbcomm.def
+++ b/src/common/fbbcomm.def
@@ -493,9 +493,11 @@
       (OPTIONAL, "int", "flag"),
       # absolute filename where (if) the library was found
       (OPTIONAL, STRING, "absolute_filename"),
-      # dlopen() does not set errno. We don't have access to an error code,
-      # but to an error string.
-      (OPTIONAL, STRING, "error_string"),
+      # dlopen() does not set errno. Also, as per #920, we cannot get
+      # the error string in a simple way without altering the
+      # intercepted process's behavior and without nasty hacks. So let's
+      # just go with a simple boolean denoting success vs. failure.
+      (REQUIRED, "bool", "error"),
     ]),
 
     ("memfd_create", [

--- a/src/firebuild/process_fbb_adaptor.cc
+++ b/src/firebuild/process_fbb_adaptor.cc
@@ -35,7 +35,7 @@ int ProcessFBBAdaptor::handle(Process *proc, const FBBCOMM_Serialized_freopen *m
 
 int ProcessFBBAdaptor::handle(Process *proc, const FBBCOMM_Serialized_dlopen *msg, int fd_conn,
                              const int ack_num) {
-  if (!msg->has_error_string() && msg->has_absolute_filename()) {
+  if (!msg->get_error() && msg->has_absolute_filename()) {
     return proc->handle_open(AT_FDCWD,
                              msg->get_absolute_filename(), msg->get_absolute_filename_len(),
                              O_RDONLY, 0, -1, 0, fd_conn, ack_num, false, false);

--- a/src/interceptor/tpl_dlopen.c
+++ b/src/interceptor/tpl_dlopen.c
@@ -7,7 +7,7 @@
 ### extends "tpl.c"
 
 {% set msg_add_fields = ["if (absolute_filename != NULL) BUILDER_SET_ABSOLUTE_CANONICAL(" + msg + ", absolute_filename);",
-                         "if (!success) fbbcomm_builder_dlopen_set_error_string(&ic_msg, dlerror());"] %}
+                         "fbbcomm_builder_dlopen_set_error(&ic_msg, !success);"] %}
 
 ### block before
   thread_libc_nesting_depth++;
@@ -17,12 +17,18 @@
   thread_libc_nesting_depth--;
 
   char *absolute_filename = NULL;
-  if (ret != NULL) {
+  if (success) {
     struct link_map *map;
     if (dlinfo(ret, RTLD_DI_LINKMAP, &map) == 0) {
       /* Note: contrary to the dlinfo(3) manual page, this is not necessarily absolute. See #657.
        * We'll resolve to absolute when setting the FBB field. */
       absolute_filename = map->l_name;
+    } else {
+      /* As per #920, dlinfo() returning an error _might_ cause problems later on in the intercepted
+       * app, should it call dlerror(). A call to dlerror() would return a non-NULL string
+       * describing dlinfo()'s failure, rather than NULL describing dlopen()'s success. But why
+       * would any app invoke dlerror() after a successful dlopen()? Let's hope that in practice no
+       * application does this. */
     }
   }
 ### endblock after

